File: scala/src/HelloLoop.scala
object HelloLoop {
  def main(args: Array[String]): Unit = {
    var worldArray = Array("a", "b", "c");

    worldArray.foreach { world =>
      println(s"world is ${world}!")
    }

    var worldMap = Map("a" -> "aaa", "b" -> "bbb", "c" -> "ccc")

    worldMap.foreach { case (key, value) =>
      println(s"world ${key} is ${value}!");
    }
  }
}

File: java/src/HelloLoop.java
import java.util.ArrayList;
import java.util.HashMap;
import java.text.MessageFormat;

/**
 * Say hello.
 * @author RyoWakabayashi
 * @version 1.0
 */
public final class HelloLoop {
  private HelloLoop() {
    throw new IllegalStateException("Utility class");
  }

  /**
    * main.
    * @param args Dummy
    */
  public static void main(final String[] args) {
    ArrayList<String> worldArray = new ArrayList<String>();
    worldArray.add("a");
    worldArray.add("b");
    worldArray.add("c");

    String format = "world is {0}!";
    for (String world: worldArray) {
      System.out.println(MessageFormat.format(format, world));
    }

    HashMap<String, String> worldMap = new HashMap<String, String>();
    worldMap.put("a", "aaa");
    worldMap.put("b", "bbb");
    worldMap.put("c", "ccc");

    String mapFormat = "world {0} is {1}!";
    for (HashMap.Entry<String, String> entry: worldMap.entrySet()) {
      System.out.println(
        MessageFormat.format(mapFormat, entry.getKey(), entry.getValue()));
    }
  }
}
